### ISSUE TITLE
Issue creating "prometheus" container on cluster

### ISSUE THREAD

**Issue by shaunthium** [coreos/prometheus-operator#74]
Hi,

I was trying out the new `v0.2.0` release of the Prometheus Operator on my `minikube` cluster this morning with kubernetes v1.5 and everything worked beautifully. Thank you guys for all the work! However, when I tried the same on my GCE cluster, I got an error when trying to create the `StatefulSet` in charge of creating the `prometheus` pods.

My events log for the `StatefulSet` pod is as follows:
```
pet: prometheus-k8s-0

Successfully assigned prometheus-k8s-0 to <redacted>

Container image "quay.io/prometheus/prometheus:v1.4.1" already present on machine

warning
Failed to create docker container "prometheus" of pod "prometheus-k8s-0_default(<redacted>)" with error: Error response from daemon: no such file or directory

pulling image "jimmidyson/configmap-reload"

Successfully pulled image "jimmidyson/configmap-reload"

Created container with docker id <redacted>; Security:[seccomp=unconfined]

Started container with docker id <redacted>

Created container with docker id <redacted>; Security:[seccomp=unconfined]

Started container with docker id <redacted>
```

The `configmap-reload` image seems to creating fine, but the `prometheus` image errors out with the message above. As a result, my `StatefulSet` is unable to be created.

The YAML file for my Prometheus TPR is as follows:
```
apiVersion: monitoring.coreos.com/v1alpha1
kind: Prometheus
metadata:
  name: prometheus-k8s
  labels:
    prometheus: k8s
spec:
  version: v1.4.1
  resources:
    requests:
      memory: 400Mi
```
In short, I'm able to get past this step on my local `minikube` environment, but my GCE cluster throws the error above. Any and all assistance would be greatly appreciated! Thank you.

**Comment by brenix** [comment]
I also tried the new version (0.2.0) with k8s 1.5.1 (bare-metal) and see the same issue.

**Comment by brancz** [comment]
Very interesting, I'm investigating.

**Comment by brancz** [comment]
It seems that I can reproduce this, for some further investigation can you both tell me:

* how you created your clusters (manually, kops, gke, bootkube, tectonic-installer)
* operating system
* output of `kubectl version`
* docker version used

Thanks for reporting!

**Comment by shaunthium** [comment]
Cluster: `gke`
OS: `CoreOS 1185.5.0`
`kubectl version`: 
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:57:05Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:52:01Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```
Docker version: `1.11.2`

Thanks @brancz !

**Comment by brenix** [comment]
Mine is pretty much the same
Cluster setup: CoreOS bare-metal PXE via ignition configs
OS: CoreOS 1185.5.0
Docker Version: 1.11.2
Kubectl Version:
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:57:05Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1+coreos.0", GitCommit:"cc65f5321f9230bf9a3fa171155c1213d6e3480e", GitTreeState:"clean", BuildDate:"2016-12-14T04:08:28Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by brancz** [comment]
It seems that it is the `subPath` field of an `emptyDir` Volume. I put together a "minimal failing" example.

```
apiVersion: v1
kind: Pod
metadata:
  name: busybox
spec:
  containers:
  - image: busybox
    name: busybox
    command:
      - sleep
      - "3600"
    volumeMounts:
    - mountPath: /var/test/data
      name: test-data
      subPath: some-path
  volumes:
  - name: test-data
    emptyDir: {}
```

When removing the `subPath` it works. Investigating how we can solve this problem. Likely the best fix is to just leave out the `subPath` for `emptyDir` volumes as that will be non disruptive for those and it will still be present for all other "real" volumes. Testing the fix and will submit it soon if it doesn't turn out to be problematic.

**Comment by brancz** [comment]
fyi I have opened #75 and would appreciate everyone with a PV provisioner (to be declared in the Prometheus spec under [`storage`](https://github.com/coreos/prometheus-operator/blob/8e17baa9bb5d68b336772b601ca40de3706fbee6/Documentation/prometheus.md#storagespec)) to try out the version from the PR so we can make sure there are no further issues with other volume providers.

**Comment by brenix** [comment]
@brancz: I have tested your copy of the prometheus-operator image (tag: 5125060), which I assume includes the fix from #75. I can confirm that subPath is no longer in the statefulset and the container was able to start successfully this time. Thanks for turning this around quickly!

**Comment by shaunthium** [comment]
@brancz I can also confirm that your fix works. Thank you!

**Comment by brancz** [comment]
Thanks for reporting and testing. I will prepare a release and we should be good to go.
